### PR TITLE
Xie8899 8.0 update timesheet activity report

### PR DESCRIPTION
**Pull request by victor-ditay**

### CODE REVIEW

**Comment by victor-ditay** [comment]
To replace #109 Please have a check

Please have a check @noahzaozao @elicoidal @seb-elico

**Comment by elicoidal** [comment]
Travis shoudl be green

**Comment by victor-ditay** [comment]
> Travis should be green

@elicoidal Yes but better let be fix master branch by #154 and rebase this

**Comment by seb-elico** [comment]
@victormartinelicocorp I don't understand why you want to replace the original PR #109. The repo is public, so is the fork of XP. So you should be able to push to his branch so that it updates the original PR! And we won't lose all the previous comments. Any reason why?

**Review by elicoidal — APPROVED** [review]

File: project_task_category/README.rst [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+ * have basic modules installed (project)
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/Elico-Corp/odoo/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/Elico-Corp/odoo/issues/new?body=module:%20project_task_category%0Aversion:%20{8.0}%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (elicoidal) [inline project_task_category/README.rst:24]: remove

File: project_task_category/README.rst [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+* Rona Lin <rona.lin@elico-corp.com>
+
+Maintainer
+----------
+
+.. image:: https://www.elico-corp.com/logo.png
+:alt: Elico Corp
+:target: https://www.elico-corp.com
+
+This module is maintained by Elico Corporation.
```
Comment (elicoidal) [inline project_task_category/README.rst:41]: update

**Review by elicoidal — APPROVED** [review]
small details

**Comment by elicoidal** [comment]
@seb-elico we probably do not have access anymore to original repo

File: business_requirement_activity_report/README.rst [hunk 3]
```diff
@@ -2,9 +2,9 @@
    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
    :alt: License: AGPL-3
 
-=========================
-Timesheet Activity Report
-=========================
+====================================
+Business Requirement Activity Report
```
Comment (noahzaozao) [inline business_requirement_activity_report/README.rst:6]: why the name change to Business Requirement Activity Report , so it should be in the OCA business requirement repo ?

File: business_requirement_activity_report/report/timesheet_activity_report.py [hunk 4]
```diff
@@ -1,126 +1,144 @@
...
+        help="Activity Stage")
+    account_id = fields.Many2one(
+        comodel_name='account.analytic.account',
+        string='Analytic account',
+        readonly=True)
+    activity_id = fields.Char(
+        string='Activity id', readonly=True, help="Task id or Issue id")
+    activity_name = fields.Char(
+        string='Activity name', readonly=True, help="Task name or Issue name"),
+    task_categ_id = fields.Many2one(
```
Comment (noahzaozao) [inline business_requirement_activity_report/report/timesheet_activity_report.py:56]: be careful with this field , we did migration of elico 8 , I've commented this fields in many modules.

File: business_requirement_activity_report/report/timesheet_activity_report.py [hunk 5]
```diff
@@ -1,126 +1,144 @@
...
+        readonly=True,
+        help="Task category")
+    br_id = fields.Many2one(
+        comodel_name='business.requirement',
+        string='Bus. requ.',
+        readonly=True,
+        help="Business requirement")
+    partner_id = fields.Many2one(
+        comodel_name='res.partner', string='Customer', readonly=True)
+    project_categ_id = fields.Many2one(
```
Comment (noahzaozao) [inline business_requirement_activity_report/report/timesheet_activity_report.py:68]: not sure this field should be project.category or project.project.category

File: business_requirement_activity_report/report/timesheet_activity_report.py [hunk 6]
```diff
@@ -1,126 +1,144 @@
...
+                        al.date,
+                        p.id AS project_id,
+                        p.state AS project_state,
+                        COALESCE(t.stage_id, i.stage_id) AS activity_stage_id,
+                        COALESCE(a.id, al.account_id) AS account_id,
+                        COALESCE(t.id, i.id) AS activity_id,
+                        COALESCE(t.name, i.name) AS activity_name,
+                        -- The hours are null in case the timesheet
+                        -- is not linked to a task
+                        t.task_categ_id,
```
Comment (noahzaozao) [inline business_requirement_activity_report/report/timesheet_activity_report.py:120]: be careful with this field , we commented this field during the migration

File: business_requirement_activity_report/report/timesheet_activity_report_view.xml [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+            <tree string="Timesheet Activities Tree">
+                <field name="partner_id"/>
+                <field name="account_id"/>
+                <field name="project_categ_id"/>                    
+                <field name="project_id"/>
+                <field name="project_state"/>
+                <field name="br_id"/>
+                <field name="activity_id" string="ID"/>
+                <field name="activity_type"/>
+                <field name="task_categ_id"/>
```
Comment (noahzaozao) [inline business_requirement_activity_report/report/timesheet_activity_report_view.xml:17]: be careful with this field , we commented this field during the migration

File: business_requirement_activity_report/report/timesheet_activity_report_view.xml [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+            <search string="Timesheet search">
+                <field name="activity_id"/>
+                <field name="account_id"/>
+                <field name="project_id"/>
+                <field name="project_state"/>
+                <field name="user_id"/>
+                <field name="br_id"/>
+                <field name="partner_id"/>
+                <field name="activity_stage_id"/>
+                <field name="task_categ_id"/>
```
Comment (noahzaozao) [inline business_requirement_activity_report/report/timesheet_activity_report_view.xml:64]: be careful with this field , we commented this field during the migration

File: business_requirement_activity_report/report/timesheet_activity_report_view.xml [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+                <separator/>
+                <group expand="0" string="Group By">
+                    <filter string="Analytic Account" domain="[]" context="{'group_by': 'account_id'}"/>
+                    <filter string="Project" domain="[]" context="{'group_by': 'project_id'}"/>
+                    <filter string="State" domain="[]" context="{'group_by': 'project_state'}"/>
+                    <filter string="User" domain="[]" context="{'group_by': 'user_id'}"/>
+                    <filter string="Business Requirement" domain="[]" context="{'group_by': 'br_id'}"/>
+                    <filter string="Customer" domain="[]" context="{'group_by': 'partner_id'}"/>
+                    <filter string="Activity stage" domain="[]" context="{'group_by': 'activity_stage_id'}"/>
+                    <filter string="Task category" domain="[]" context="{'group_by': 'task_categ_id'}"/>
```
Comment (noahzaozao) [inline business_requirement_activity_report/report/timesheet_activity_report_view.xml:79]: be careful with this field , we commented this field during the migration

File: project_task_category/models/task_category.py [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+    _sql_constraints = [
+        ('name_uniq', 'unique (name)',
+         'Name already exists.'),
+    ]
+
+
+class ProjectTask(models.Model):
+    _inherit = 'project.task'
+
+    task_categ_id = fields.Many2one(
```
Comment (noahzaozao) [inline project_task_category/models/task_category.py:26]: be careful with this field , we commented this field during the migration , maybe it should be 'project.category' ?

File: business_requirement_activity_report/README.rst (hunk 3 above)
Comment (martintorresvm) [reply business_requirement_activity_report/README.rst:6]: > To replace #109 Please have a check
Please have a check @noahzaozao @elicoidal @seb-elico

Not really idea I continue the left PR created @xie8899 to solve merge problems and not sure about this maybe @elicoidal could reply or you could have a look on previous history of comments.

File: business_requirement_activity_report/report/timesheet_activity_report.py (hunk 4 above)
Comment (martintorresvm) [reply business_requirement_activity_report/report/timesheet_activity_report.py:56]: Yes I would use asap this field related to a new module we recently proposed to https://github.com/OCA/project/pull/225 or just drop it to merge this PR the sooner the better.

**Comment by seb-elico** [comment]
@noahzaozao thanks for the warning about `task_categ_id`. Indeed, we commented it out because right now, the field is in a tag. It should be solved very soon ( @victormartinelicocorp is working on it), so let's wait for that before we merge this PR.

File: business_requirement_activity_report/README.rst (hunk 3 above)
Comment (seb-elico) [reply business_requirement_activity_report/README.rst:6]: @noahzaozao this module is very specific, it depends on way too many modules to be included in the OCA BR repo. Definitely belongs to odoo-addons

**Comment by elicoidal** [comment]
@seb-elico @victormartinelicocorp Let's move this topic asap now

**Comment by seb-elico** [comment]
@victormartinelicocorp can you make sure Travis is building please? XML files don't seem to be formatted very well

**Comment by elicoidal** [comment]
will be replaced by new API clean version

**Pull request closed without merging**